modules/boltlib/lib/puppet/functions/get_targets.rb

```diff
@@ -0,0 +1,32 @@
+# Parses common ways of referring to targets and returns an array of Targets.
+#
+# Accepts input consisting of
+# - a group
+# - a target URI
+# - an array of groups and/or target URIs
+# - a string that consists of a comma-separated list of groups and/or target URIs
+#
+# Examples of the above would be
+# - 'group1'
+# - 'host1,group1,winrm://host2:54321'
+# - ['host1', 'group1', 'winrm://host2:54321']
+#
+# Returns an array of Targets resolved from any target URIs and groups.
+
+require 'bolt/error'
+
+Puppet::Functions.create_function(:get_targets, Puppet::Functions::InternalFunction) do
+  local_types do
```

Comment:
I think this should be a public type now so it can be used in plan signatures. Maybe we should bikeshed the name first...

Reply:
Yeah. Also, what's all the `Puppet::Functions::InternalFunction` stuff?